### PR TITLE
setup delayed deployments

### PR DESCRIPTION
**Pull request by ivanzhelyazkov**
Setup delayed deployment role configuration for CarbonController, Voucher and FeeBurner.

### CODE REVIEW

**Review by yudilevi — APPROVED** [review]

File: deploy/scripts/000100-revoke-roles.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    // Grant CarbonController admin roles to dao multisig
+    await grantRole({
+        name: InstanceName.CarbonController,
+        id: Roles.Upgradeable.ROLE_ADMIN,
+        member: daoMultisig,
+        from: deployer
+    });
+
+    await revokeRole({
```
Comment (lbeder) [inline deploy/scripts/000100-revoke-roles.ts:24]: Nitpick: you can call `renounceRole` to be a bit more explicit

File: deploy/scripts/000100-revoke-roles.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        name: InstanceName.FeeBurner,
+        id: Roles.Upgradeable.ROLE_ADMIN,
+        member: deployer,
+        from: deployer
+    });
+
+    // renounce CarbonController's ROLE_EMERGENCY_STOPPER role from the deployer
+    await renounceRole({
+        name: InstanceName.CarbonController,
+        id: Roles.CarbonController.ROLE_EMERGENCY_STOPPER,
```
Comment (lbeder) [inline deploy/scripts/000100-revoke-roles.ts:64]: Perhaps the DAO multisig should be set here explicitly as well?

File: deploy/scripts/000101-transfer-proxy-admin-ownership.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { execute, InstanceName, isLive, setDeploymentMetadata } from '../../utils/Deploy';
+import { DeployFunction } from 'hardhat-deploy/types';
+import { HardhatRuntimeEnvironment } from 'hardhat/types';
+
+const func: DeployFunction = async ({ getNamedAccounts }: HardhatRuntimeEnvironment) => {
+    const { deployer, daoMultisig } = await getNamedAccounts();
+
+    await execute({
+        name: InstanceName.ProxyAdmin,
+        methodName: 'transferOwnership',
```
Comment (lbeder) [inline deploy/scripts/000101-transfer-proxy-admin-ownership.ts:10]: We should consider an `Ownable2Step`-based (https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/access/Ownable2Step.sol) `ProxyAdmin`

**Review by lbeder — APPROVED** [review]

File: deploy/scripts/000100-revoke-roles.ts (hunk 2 above)
Comment (ivanzhelyazkov) [reply deploy/scripts/000100-revoke-roles.ts:64]: Discussed with Yudi to not set it in the scripts

File: deploy/scripts/000101-transfer-proxy-admin-ownership.ts (hunk 3 above)
Comment (ivanzhelyazkov) [reply deploy/scripts/000101-transfer-proxy-admin-ownership.ts:10]: I agree, even though this isn't in the PR's scope

File: deploy/scripts/000100-revoke-roles.ts (hunk 1 above)
Comment (ivanzhelyazkov) [reply deploy/scripts/000100-revoke-roles.ts:24]: I prefer `revokeRole` actually for the admin roles, makes for a bit more readable code.

File: deploy/scripts/000100-revoke-roles.ts (hunk 2 above)
Comment (lbeder) [reply deploy/scripts/000100-revoke-roles.ts:60]: 👍

File: deploy/scripts/000101-transfer-proxy-admin-ownership.ts (hunk 3 above)
Comment (lbeder) [reply deploy/scripts/000101-transfer-proxy-admin-ownership.ts:10]: 👍 

@yudilevi FYI

**Pull request merged**